### PR TITLE
Planning

### PR DESCRIPTION
**Pull request by frustak**
Related to #9 
For now, we can add to-dos to goals, other parts are WIP.

### CODE REVIEW

File: src/features/goal/list.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
 import { Eraser } from "phosphor-react";
 import { trpc } from "../../utils/trpc";
 import { IconButton } from "../ui/button";
+import { Loader } from "../ui/loader";
 
-type Goal = { id: string; name: string };
+export type Goal = { id: string; name: string };
```
Comment (Glyphack) [inline src/features/goal/list.tsx]: Can't we somehow get this from the defined types in trpc?

File: src/features/todo/form.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  });
+
+  return (
+    <form onSubmit={onSubmit} className="flex">
+      <div className="flex items-stretch rounded-sm border-2 border-neutral-700">
+        <input
+          placeholder="What you want to do?"
+          autoFocus
+          className="grow px-2 py-0.5 focus:outline-0"
+          {...form.register("title")}
```
Comment (Glyphack) [inline src/features/todo/form.tsx:45]: Don't we need description here?

File: src/features/todo/list.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { trpc } from "../../utils/trpc";
+import { Loader } from "../ui/loader";
+
+export const TodosList = ({ goalId }: { goalId: string }) => {
+  const todosQuery = trpc.todo.getTodos.useQuery({ goalIds: [goalId] });
+  const todos = todosQuery.data?.todos ?? [];
+
+  if (todosQuery.isLoading) return <Loader />;
+  if (todos.length === 0) return <p>No todos</p>;
```
Comment (Glyphack) [inline src/features/todo/list.tsx]: We can replace this with:

Nothing to do but chill

Comment (Glyphack) [reply src/features/todo/list.tsx]: Be frusty 🥶

File: src/features/todo/list.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  return (
+    <ul className="list-inside list-disc space-y-3">
+      {todos.map((todo) => (
+        <li key={todo.id}>{todo.title}</li>
```
Comment (Glyphack) [inline src/features/todo/list.tsx:15]: We can have a small text for todo description.

**Review by Glyphack — CHANGES_REQUESTED** [review]

File: src/features/todo/form.tsx (hunk 2 above)
Comment (frustak) [reply src/features/todo/form.tsx:45]: I've tried a minimal implementation, maybe we can add description in another PR?

File: src/features/todo/list.tsx (hunk 4 above)
Comment (frustak) [reply src/features/todo/list.tsx:15]: Yep, I'll fix it in the next PR :P

File: src/features/goal/list.tsx (hunk 1 above)
Comment (frustak) [reply src/features/goal/list.tsx]: I've tried to use trpc schema to infer type, it's a good idea to do it for other schemas too

**Review by Glyphack — APPROVED** [review]

**Comment by Glyphack** [comment]
LGTM!

**Pull request merged**